aten/src/ATen/native/cuda/ForeachBinaryOpScalarList.cu

```diff
@@ -67,8 +68,53 @@ std::vector<Tensor> foreach_tensor_##NAME##_scalarlist_kernel_cuda(TensorList te
 }
 
 FOREACH_BINARY_OP_SCALARLIST(add, std::plus);
-FOREACH_BINARY_OP_SCALARLIST(sub, std::minus);
 FOREACH_BINARY_OP_SCALARLIST(mul, std::multiplies);
-FOREACH_BINARY_OP_SCALARLIST(div, std::divides);
+
+// In the case of division, integer inputs will result in float. 
+// Currently multi tensor apply can only return result of the same type as input.
+void foreach_tensor_div_scalarlist_kernel_cuda_(TensorList tensors, at::ArrayRef<Scalar> scalars) {
```

Comment:
Also nit: is it possible to update FOREACH_BINARY_OP_SCALARLIST to take three arguments, where the last argument is div_op? Then we would be able to use FOREACH_BINARY_OP_SCALARLIST with division instead of rewriting the division logic

Reply:
Done